### ISSUE TITLE
Metadata: Allow indexes and foreign keys to use inherited properties

### ISSUE THREAD

**Issue by AndriySvyryd** [aspnet/EntityFramework#2514]
The ModelBuilder should lift the FK or index when there is one declared on the same properties on a derived type

**Comment by rowanmiller** [comment]
@AndriySvyryd if you don't think this is critical for 7.0.0 then come discuss as we may postpone it

**Comment by AndriySvyryd** [comment]
7.0.0 and pri0 are appropriate for this

**Comment by rowanmiller** [comment]
@AndriySvyryd can you add some more details on what this means?

**Comment by AndriySvyryd** [comment]
Consider
```
class Base
{
  public int Id { get; set; }
}
class Derived : Base
{
  public string Name { get; set; }
}
```

You should be able to create an index like this:
```
modelBuilder.Entity<Derived>().Index(b => new [] { b.Id, b.Name });
```

Also if you had created an index on the derived entity type:
```
  modelBuilder.Entity<Derived>().Index(b => new [] { b.Id });
```

You should be able to override it on base:
```
  modelBuilder.Entity<Base>().Index(b => new [] { b.Id });
```
and viceversa.
Same applies to FKs.

**Comment by rowanmiller** [comment]
@AndriySvyryd can you discuss with folks what we want to do before doing it. Either design meeting, or just in the team room.

**Comment by smitpatel** [comment]
Partially implementing this for indexes alongwith #700

**Comment by smitpatel** [comment]
for indexes implemented in bc4609bcb34f57325635e3f3c0a1c20606509cd5
Pending for foreignkeys

**Comment by smitpatel** [comment]
Should base type lift the FK from derived type if defined on same features? or block it?
